oneargopy/Argo.py

```diff
@@ -284,18 +283,51 @@ def trajectories(self, floats: int | list | dict)-> None:
         # Displaying graph
         plt.show()
 
-    def load_float_data(self, floats: int | list)-> None: 
+    def load_float_data(self, floats: int | list | dict, parameters: str | list = None)-> pd: 
         """ A function to load float data into memory.
-        """
 
+            :param: floats : int | list | dict - A float or list of floats to  
+                load data from. Or a dictionary specifying floats and profiles
+                to read from the .nc file. 
+            :param: parameters : str | list - An optional parameter to list parameters
+                that the user would like included in the dataframe. If the parameter is not
+                in the float passed then only the surface level of the profile will be included.
+
+            :return: float_data : pd - A dataframe with requested float data. 
+        """
         # Check that index files are in memory
+        if not self.download_settings.keep_index_in_memory: 
+            self.sprof_index = self.__load_sprof_dataframe()
+            self.prof_index = self.__load_prof_dataframe()
 
         # Check that passed float is inside of the dataframes
+        self.float_ids = floats
+        self.__validate_floats_kwarg()
 
-        # Download .nc files for passed floats
+        # Validate passed parameters
+        self.float_parameters = parameters
+        if self.float_parameters : self.__validate_float_parameters_arg()
 
-        # 
-        pass
+        # Download .nc files for passed floats
+        files = []
+        for wmoid in self.float_ids : 
+            # Generate filename 
+            ## If the float is a bgc float it will have a corresponding sprof file
+            if self.float_stats.loc[self.float_stats['wmoid'] == wmoid, 'is_bgc'].values[0] : 
+                file_name = f'{wmoid}_Sprof.nc'
+                files.append(file_name)
+            ## If the float is a phys float it will have a corresponding prof file
+            else :
```

Comment:
please replace "blank colon" with "colon" in all cases like these (don't replace all at once, in some cases there should be a blank before the colon, mostly for indentation)